src/InteractiveWindow/Editor/InteractiveWindow_UIThread.cs

```diff
@@ -214,12 +213,13 @@ public Task<ExecutionResult> ResetAsync(bool initialize)
                     _window._currentLanguageBuffer = null;
                 }
 
-                // replace the task being interrupted by a "reset" task:
                 State = State.Resetting;
-                _currentTask = _window._evaluator.ResetAsync(initialize);
-                _currentTask.ContinueWith(FinishExecute, _uiScheduler);
+                var executionResult = await _window._evaluator.ResetAsync(initialize).ConfigureAwait(true);
```

Comment:
Oh I don't mean assert that you're on the UI thread after the await. Assert you're on the UI thread at the start of the method. On the language service side of things we have a `ForegroundThreadAffinitizedObject` that offers convenient helpers for this.